### PR TITLE
fix staticcheck in k8s.io/apiserver/pkg/storage/value/encrypt/envelope

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Fix the static check issue ([SA2002](https://staticcheck.io/docs/checks#SA2002)) in k8s.io/apiserver/pkg/storage/value/encrypt/envelope:
```bash
Errors from staticcheck:
vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:136:4: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:142:6: call to T.Fatalf
vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:151:4: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
vendor/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:160:6: call to T.Fatalf

```
#### Which issue(s) this PR fixes:

Part of #92402

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
/approve
for hack/

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 1]
```diff
@@ -127,6 +127,7 @@ func TestTimeouts(t *testing.T) {
 				kmsPluginWG     sync.WaitGroup
 				testCompletedWG sync.WaitGroup
 				socketName      = newEndpoint()
+				errc            = make(chan error, 1)
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:130]: if we use channel to propagate setup error then we don't need `kubeAPIServerWG` and  `kmsPluginWG` any longer.
```
kubeAPIErrCh, pluginErrCh   = make(chan error, 1), make(chan error, 1)
```

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 2]
```diff
@@ -139,7 +140,8 @@ func TestTimeouts(t *testing.T) {
 
 				service, err = NewGRPCService(socketName.endpoint, tt.callTimeout)
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:141]:
```
if err != nil {
    kubeAPIErrCh <- fmt.Errorf("failed to create envelope service, error: %v", err)
    return
}
kubeAPIErrCh <- nil
```

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 3]
```diff
@@ -154,30 +156,42 @@ func TestTimeouts(t *testing.T) {
 
 				f, err := mock.NewBase64Plugin(socketName.path)
 				if err != nil {
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:158]: use `pluginErrCh` here as shown above

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go [hunk 4]
```diff
@@ -154,30 +156,42 @@ func TestTimeouts(t *testing.T) {
...
 				}
 				defer f.CleanUp()
 				kmsPluginWG.Done()
 				// Keeping plugin up to process requests.
 				testCompletedWG.Wait()
 			}()
 
-			kubeAPIServerWG.Wait()
-			_, err = service.Encrypt(data)
+			go func() {
```
Comment (tkashem) [inline staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:172]: we don't need this go routine here:
```
if err := <-kubeAPIErrCh; err != nil {
     t.Fatal("Expected envelop service to be ready, but got: %v", err)
}

_, err = service.Encrypt(data)

if err == nil && tt.wantErr != "" {
     t.Fatalf("got nil, want %s", tt.wantErr)
}
if err != nil && tt.wantErr == "" {
    t.Fatalf("got %q, want nil", err.Error())
}

if err := <-pluginErrCh; err != nil {
     t.Fatal("Expected kms plugin to be ready, but got: %v", err)
}
```

File: staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go (hunk 1 above)
Comment (haoruan) [reply staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go:130]: make sense

**Comment by tkashem** [comment]
looks good to me, just as a precaution can you run this test with `stress` and make sure that it does not have any flakes?

```
go test -race k8s.io/apiserver/pkg/storage/value/encrypt/envelope -c
stress ./envelope.test -test.run=TestTimeouts
```

https://pkg.go.dev/golang.org/x/tools/cmd/stress

**Comment by haoruan** [comment]
No failure for 1 hour:
```
1h0m0s: 4499173 runs so far, 0 failures
```

**Comment by tkashem** [comment]
/lgtm

Thanks!

**Comment by dims** [comment]
please rebase, we no longer have the hack/.staticcheck_failures file.

**Review by aramase — COMMENTED** [review]
With #106448, static check for `SA2002` is being excluded and not checked
https://github.com/kubernetes/kubernetes/blob/f538b0b105552abeb2c52a3cc3650c108acb1429/.golangci.yaml#L36

`pull-kubernetes-verify` in the PR gate isn't actually checking if the changes fix the check. Should we wait until the check is enabled before merging this PR?

**Pull request closed without merging**